src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -0,0 +1,445 @@
+                                                            parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7));
+            Assert.True(compilation2.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            m1 = compilation2.GetMember<MethodSymbol>("I1.M1");
+            var test2 = compilation2.GetTypeByMetadataName("Test2");
+
+            Assert.Same(m1, test2.FindImplementationForInterfaceMember(m1));
+
+            compilation2.VerifyDiagnostics();
+            CompileAndVerify(compilation2, verify: false);
+        }
+
+        [Fact]
+        public void MethodImplementation_08()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    I1 M1() 
+    {
+        throw null;
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, new[] { SystemCoreRef }, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.False(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+
+            var m1 = compilation1.GetMember<MethodSymbol>("I1.M1");
+
+            Assert.False(m1.IsAbstract);
+            Assert.True(m1.IsVirtual);
+
+            compilation1.VerifyDiagnostics(
+                // (4,8): error CS8401: Target runtime doesn't support default interface implementation.
+                //     I1 M1() 
+                Diagnostic(ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementation, "M1").WithLocation(4, 8)
+                );
+
+            Assert.True(m1.IsMetadataVirtual());
+
+            Assert.Throws<System.InvalidOperationException>(() => compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation = false);
+        }
+
+        [Fact]
+        public void MethodImplementation_09()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    static void M1() 
+    {
+        System.Console.WriteLine(""M1"");
+    }
+}
+
+class Test1 : I1
+{}
+";
+            var compilation1 = CreateCompilationWithMscorlib(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+
+            var m1 = compilation1.GetMember<MethodSymbol>("I1.M1");
+
+            Assert.False(m1.IsAbstract);
+            Assert.True(m1.IsVirtual);
+            Assert.False(m1.IsStatic);
+
+            var test1 = compilation1.GetTypeByMetadataName("Test1");
+
+            Assert.Same(m1, test1.FindImplementationForInterfaceMember(m1));
+
+            compilation1.VerifyDiagnostics(
+                // (4,17): error CS0106: The modifier 'static' is not valid for this item
+                //     static void M1() 
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "M1").WithArguments("static").WithLocation(4, 17),
```

Comment:
No one is making a claim that the entire spec is implemented. Please focus on the subject of this PR. Private or static members are not supported in interfaces at all. The behavior is correct given the current state of the product.